### PR TITLE
LPS-164449

### PR DESCRIPTION
**Pull request by sontruongces**
Hi team,

This is the PR for the [LPS-164449](https://issues.liferay.com/browse/LPS-164449), the original has passed my review at sontruongces#47. Please help me to review and give feedback.

Thanks
cc: @thienquach87

### CODE REVIEW

**Comment by thienquach87** [comment]
ci:test:sf

**Comment by thienquach87** [comment]
Currently, in case HSQL is used in liferay-portal as a Database, if the createdDate and modifiedDate are used to filture, there is an exception thrown "invalid datetime...". The reason is because the current date format is not supported in HSQL. To fix this issue I have solution as below:
 - Setting sql.syntax_ora=true to enable support for non-standard syntax of Oracle.
- Because HSQL only support four date's patterns, 
                        pattern = "DD-MON-YY";
                        pattern = "DD-MON-YYYY";
                        pattern = "DD-MON-YYYY HH24:MI:SS";
                        pattern = "DD-MON-YYYY HH24:MI:SS.FF";
so I convert the date fortmat to the pattern supported by HSQL.

**Comment by thienquach87** [comment]
ci:test:relevant

**Comment by thienquach87** [comment]
Hi @MarinhoFeliphe,
Could you help me to review this PR?

**Comment by MarinhoFeliphe** [comment]
Hi @thienquach87,

Our team is aware about this pull request, we're managing who will review the pull requests in our queue, someone in BPM team is going to review this one as soon as they can.

Thank you for your contribution @sontruongces

Best regards,
Feliphe

**Comment by thienquach87** [comment]
> 

Thank @MarinhoFeliphe for letting me know.

File: modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/odata/filter/expression/PredicateExpressionVisitorImpl.java [hunk 1]
```diff
@@ -257,6 +267,39 @@ private Predicate _contains(Object fieldName, Object fieldValue) {
...
+		}
+	}
+
+	private String _getDriverName() {
+		try {
+			Connection connection = DataAccess.getConnection();
+
+			DatabaseMetaData metaData = connection.getMetaData();
+
+			String driverName = metaData.getDriverName();
```
Comment (gabrielwas) [inline modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/odata/filter/expression/PredicateExpressionVisitorImpl.java:292]: Could you use 

```
String databaseProductName = GetterUtil.getString(metaData.getDatabaseProductName());
```

and check in the row 325 with 

```
StringUtil.startsWith(_getDriverName(), "HSQL")
```

please ?

File: modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/odata/filter/expression/PredicateExpressionVisitorImpl.java [hunk 2]
```diff
@@ -279,6 +322,12 @@ else if (Objects.equals(BinaryExpression.Operation.OR, operation)) {
 		EntityField entityField = entityFieldsMap.get(
 			GetterUtil.getString(left));
 
+		if (_getDriverName().contains("HSQL")) {
```
Comment (gabrielwas) [inline modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/odata/filter/expression/PredicateExpressionVisitorImpl.java:325]: Could you use 

```
StringUtil.startsWith(_getDriverName(), "HSQL")
```

to check here?

File: modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/odata/filter/expression/PredicateExpressionVisitorImpl.java [hunk 3]
```diff
@@ -279,6 +322,12 @@ else if (Objects.equals(BinaryExpression.Operation.OR, operation)) {
...
+			String dateTimeString = right.toString();
+
+			right = _convertDateFormat(dateTimeString);
```
Comment (gabrielwas) [inline modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/odata/filter/expression/PredicateExpressionVisitorImpl.java:328]: Is it always going to be of date type? Shouldn't we be checking the type of this field?

File: portal-impl/src/com/liferay/portal/tools/HypersonicLoader.java [hunk 4]
```diff
@@ -95,10 +95,13 @@ public HypersonicLoader(
 		// See LEP-2927. Appending ;shutdown=true to the database connection URL
 		// guarantees that ${databaseName}.log is purged.
 
+		String connectionProperties =
```
Comment (gabrielwas) [inline portal-impl/src/com/liferay/portal/tools/HypersonicLoader.java:98]: After our review process, we should forward this code to core-infra team so they can check this part of the code

**Review by gabrielwas — CHANGES_REQUESTED** [review]
Hi @sontruongces !

First, thank you for your contribution! It's good to have it. 
I left some comments. Could you check them? 

Thx!

File: modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/odata/filter/expression/PredicateExpressionVisitorImpl.java (hunk 3 above)
Comment (thienquach87) [reply modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/odata/filter/expression/PredicateExpressionVisitorImpl.java:328]: yes, you're right. The field should be checked the type

File: modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/odata/filter/expression/PredicateExpressionVisitorImpl.java (hunk 2 above)
Comment (thienquach87) [reply modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/odata/filter/expression/PredicateExpressionVisitorImpl.java:325]: I updated it

**Comment by thienquach87** [comment]
ci:test:relevant

**Comment by thienquach87** [comment]
@gabrielwas,
I updated source based on your feed-back. Could you help to forward to core-infra team?

**Comment by thienquach87** [comment]
ci:reopen

**Comment by thienquach87** [comment]
ci:open

**Pull request closed without merging**